File: editors/cfd-editor/frontend/src/api/plugins.ts
import type { FrontendPluginBundle } from '../bindings/FrontendPluginBundle'
import type { FrontendPluginProjectState } from '../bindings/FrontendPluginProjectState'
import type { FrontendPluginState } from '../bindings/FrontendPluginState'
import { invokeCommand } from './tauriEnv'

export async function installFrontendPlugin(manifestPath: string): Promise<FrontendPluginBundle> {
  return invokeCommand<FrontendPluginBundle>('install_frontend_plugin', { manifestPath })
}

export async function listFrontendPlugins(): Promise<FrontendPluginState> {
  return invokeCommand<FrontendPluginState>('list_frontend_plugins')
}

export async function uninstallFrontendPlugin(id: string): Promise<void> {
  return invokeCommand<void>('uninstall_frontend_plugin', { id })
}

export async function installProjectFrontendPlugin(sessionId: number, manifestPath: string): Promise<FrontendPluginBundle> {
  return invokeCommand<FrontendPluginBundle>('install_project_frontend_plugin', { sessionId, manifestPath })
}

export async function listProjectFrontendPlugins(sessionId: number): Promise<FrontendPluginProjectState> {
  return invokeCommand<FrontendPluginProjectState>('list_project_frontend_plugins', { sessionId })
}

export async function uninstallProjectFrontendPlugin(sessionId: number, id: string): Promise<void> {
  return invokeCommand<void>('uninstall_project_frontend_plugin', { sessionId, id })
}

export async function setProjectFrontendPluginEnabled(sessionId: number, id: string, enabled: boolean): Promise<void> {
  return invokeCommand<void>('set_project_frontend_plugin_enabled', { sessionId, id, enabled })
}

File: editors/cfd-editor/src-tauri/src/plugins.rs
//! Tauri 前端插件的命令、持久化与路径安全边界。
//!
//! 按职责拆分（仅结构拆分，不改行为）：
//! - `manifest`：插件清单校验与 bundle 类型；
//! - `store`：全局/项目作用域持久化与路径安全；
//! - `commands`：7 个 Tauri 命令薄适配。

pub(crate) mod commands;
pub(crate) mod manifest;
pub(crate) mod store;

pub(crate) use commands::{
    install_frontend_plugin, install_project_frontend_plugin, list_frontend_plugins,
    list_project_frontend_plugins, set_project_frontend_plugin_enabled, uninstall_frontend_plugin,
    uninstall_project_frontend_plugin,
};
pub use manifest::{
    FrontendPluginBundle, FrontendPlugins, PluginScope, ProjectFrontendPlugins,
    ProjectPluginDefaults,
};
